cli/cmd/proxy/list/command_test.go

```diff
@@ -234,3 +239,86 @@ func TestFetchPods(t *testing.T) {
 		})
 	}
 }
+
+func TestEndToEnd(t *testing.T) {
+	// These regular expressions must be present in the output.
+	expected := []string{
+		"Namespace.*Name.*Type",
+		"consul.*mesh-gateway.*Mesh Gateway",
+		"consul.*terminating-gateway.*Terminating Gateway",
+		"default.*ingress-gateway.*Ingress Gateway",
+		"consul.*api-gateway.*Sidecar",
+		"default.*pod1.*Sidecar",
+	}
+
+	pods := []v1.Pod{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ingress-gateway",
+				Namespace: "default",
+				Labels: map[string]string{
+					"component": "ingress-gateway",
+					"chart":     "consul-helm",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "mesh-gateway",
+				Namespace: "consul",
+				Labels: map[string]string{
+					"component": "mesh-gateway",
+					"chart":     "consul-helm",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "terminating-gateway",
+				Namespace: "consul",
+				Labels: map[string]string{
+					"component": "terminating-gateway",
+					"chart":     "consul-helm",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "api-gateway",
+				Namespace: "consul",
+				Labels: map[string]string{
+					"api-gateway.consul.hashicorp.com/managed": "true",
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dont-fetch",
+				Namespace: "default",
+				Labels:    map[string]string{},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pod1",
+				Namespace: "default",
+				Labels: map[string]string{
+					"consul.hashicorp.com/connect-inject-status": "injected",
+				},
+			},
+		},
+	}
+	client := fake.NewSimpleClientset(&v1.PodList{Items: pods})
+
+	c, buf := getInitializedCommand(t)
+	c.kubernetes = client
+
+	out := c.Run([]string{"-A"})
+	require.Equal(t, 0, out)
+
+	actual := buf.String()
+
+	for _, expression := range expected {
```

Comment:
I *really* like this. I am definitely going to use it in the future!

Reply:
Thanks! I wanted a solution that wasn't too brittle.